src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -995,42 +999,42 @@ export default class Blockchain extends Emittery.Typed<
       }
     };
 
-    let txHashCurrentlyProcessing: string = null;
-    const transactionHash = Data.from(transaction.hash()).toString();
+    const transactionHash = transaction.hash();
+    let txHashCurrentlyProcessing: Buffer = null;
 
-    const beforeTxListener = async (tx: Transaction) => {
-      txHashCurrentlyProcessing = Data.from(tx.hash()).toString();
-      if (txHashCurrentlyProcessing == transactionHash) {
+    const beforeTxListener = (tx: Transaction) => {
+      txHashCurrentlyProcessing = tx.hash();
+      if (txHashCurrentlyProcessing.equals(transactionHash)) {
         if (keys && contractAddress) {
-          keys.forEach(async key => {
-            // get the raw key using the hashed key
-            let rawKey = await this.#database.storageKeys.get(
-              Data.from(key).toString()
-            );
-
-            vm.stateManager.getContractStorage(
-              Address.from(contractAddress).toBuffer(),
-              rawKey,
-              (err: Error, result: Buffer) => {
-                if (err) {
-                  throw err;
+          const database = this.#database;
```

Comment:
For reasons, `this.#database` is slow once typescript compiles things, so I like to cache the lookup.

Reply:
How did you measure the performance difference @davidmurdoch ?

Reply:
Don't need to. Just look at what typescript generates for our ES target. It's a function call, a Map `has` call, a branch, and a Map `get` call, IIRC.

I'd love to generate multiple builds so we can ship without these sorts of polyfills. I like the "true" private fields the syntax provides, and don't think I want to remove their use all together.

Happy to have your review on this!